quickstart-existing-stack-benchmark/run_example.md

```diff
@@ -0,0 +1,237 @@
+
+
+### Prepare an `llm-d` cluster
+
+Set up the stack for benchmarking. Since you are starting from an existing stack, you may need to restart some pods to create a clean baseline. For this simple example, if you want to compare different setups (e.g., various `epp` configurations) then you have to set up each configuration manually and rerun the example for each. 
+
+In this example, the benchmark sends requests to an `inference-gateway` endpoint. Replace `inference-gateway` with the name of the pod if you want to benchmark a single `vllm` instead.
+
+
+## Benchmarking Steps
+
+
+### 1. Prepare workload specification (profile)
+
+You will need a `yaml` specification to tell `inference-perf` how to generate the _Workload_ that would be used to benchmark your stack. `inference-perf` will generate prompts (AKA a _Data Set_) with timing (AKA _Load_).
+
+Several workload examples are available under `llm-d-benchmark/workload/profiles/inference-perf`. We demonstrate with `shared_prefix_synthetic`.
+
+<details>
+<summary>Click to view `shared_prefix_synthetic.yaml.in`</summary>
+
+For example, `shared_prefix_synthetic.yaml.in`:
+```yaml
+load:
+  type: constant
+  stages:
+  - rate: 2
+    duration: 50
+  - rate: 5
+    duration: 50
+  - rate: 8
+    duration: 50
+  - rate: 10
+    duration: 50
+  - rate: 12
+    duration: 50
+  - rate: 15
+    duration: 50
+  - rate: 20
+    duration: 50
+api:
+  type: completion
+  streaming: true
+server:
+  type: vllm
+  model_name: REPLACE_ENV_LLMDBENCH_DEPLOY_CURRENT_MODEL
+  base_url: REPLACE_ENV_LLMDBENCH_HARNESS_STACK_ENDPOINT_URL
+  ignore_eos: true
+tokenizer:
+  pretrained_model_name_or_path: REPLACE_ENV_LLMDBENCH_DEPLOY_CURRENT_TOKENIZER
+data:
+  type: shared_prefix
+  shared_prefix:
+    num_groups: 32                # Number of distinct shared prefixes
+    num_prompts_per_group: 32     # Number of unique questions per shared prefix
+    system_prompt_len: 2048       # Length of the shared prefix (in tokens)
+    question_len: 256             # Length of the unique question part (in tokens)
+    output_len: 256               # Target length for the model's generated output (in tokens)
+report:
+  request_lifecycle:
+    summary: true
+    per_stage: true
+    per_request: true
+storage:
+  local_storage:
+    path: /workspace
+```
+</details>
+
+If you want to create your own `inference-perf` profile then save your custom `yaml` specification with a `.yaml.in` suffix in the same directory.
+
+⚠️ Unless you know exactly what you are doing, you should only edit the `load` and `data` sections. `load` tell `inference-perf` how many sub-test ("_stages_") to run; each stage has a desired QPS and duration. `data` defines how to create the Data Set; the configuration parameters are different for each `type`.
+
+
+### 2. Log on into your cluster
+Run `oc login ...`
+
+
+### 3. Gather required parameters (mostly information about your `llm-d` stack)
+
+* **Namespace**:
```

Comment:
By the default, the **current** `namespace` will be used. Couldn't we make it optional?

Reply:
#288 got merged. Can you give it a try and let me know? Thanks!

Reply:
> #288 got merged. Can you give it a try and let me know? Thanks!

Still did not work for me. Uses default namespace instead of current.